### DISCUSSION TITLE
Different results of a resonant structure with same SI lengths

### DISCUSSION THREAD

**Discussion by francecc02** [NanoComp/meep#2135] [Q&A]
Hey everyone,
I'm a student, working with meep to generate a simulation of a resonant ZMW (a metal nanopore).

Recently I have generated to sets of h5 E-fields in order to check a weird result I saw.

I have been working with 100 nm palladium layer and 47 nm diameter pore. I chose as my characteristic length 50 nm, in order to keep the dimensions simple and small (Pd layer=2, diameter pore=0.94). However I have observed very little enhancement of the fields.
I then compared to a colleague of mine. They used a characteristic length of 1 um and described the structures as 0.1 and 0.047. 
In their results they see a significant enhancement in their E-field due to plasmonic resonance.

I checked for different properties and they used a resolution of 250, while I was using resolution of 65 in order to attain more accurate results. When I tried to lower my resolution to encompass 4nm in one pxl I found my simulations would stop due to NaN or Inf values.

I will leave the different results we produced here below. They are not normalized, but the difference should still be apparent.

Do you have an idea why we get such different results despite keeping the ratio between all structures the same?
Thank you for your help.
![XY-1um](https://user-images.githubusercontent.com/109351818/179215630-30b5aa61-e048-4f69-bc32-d28b135ae3f9.png)
![XZ-1um](https://user-images.githubusercontent.com/109351818/179215633-cb309977-4fbf-4159-b2a3-7c17e3bbc106.png)
![XZ-50nm](https://user-images.githubusercontent.com/109351818/179215635-11e56ee8-3a37-433d-ae94-088d46f1d85b.png)
![YZ-50nm](https://user-images.githubusercontent.com/109351818/179215638-b94cdd5a-b152-48a7-835e-eb081eea2334.png)

**Comment by stevengj** [comment]
What model are you using for the material?  Perhaps you accidentally changed the ε property when you switched units?

**Reply by francecc02** [reply]
Hey, 
I am just importing the mp.Pd for the palladium geometry.
Is there some scaling I need to do?

**Reply by oskooi** [reply]
As described in the user manual, the [materials library](https://meep.readthedocs.io/en/latest/Materials/#materials-library) (from which you are importing Pd) uses a default lengthscale of **1 μm**. If your chosen lengthscale is different (e.g., 100 nm), you will need to modify this file directly:

*Fitting parameters for all materials are defined for a unit distance of 1 µm. For simulations which use a different value for the unit distance, the predefined variable `um_scale` (Python) or um-scale (Scheme) must be scaled by multiplying by whatever the unit distance is, in units of µm. For example, if the unit distance is 100 nm, this would require adding the line `um_scale = 0.1*um_scale` after the line where [um_scale is defined](https://github.com/NanoComp/meep/blob/master/python/materials.py#L7). This change must be made directly to the materials library file.*

**Reply by francecc02** [reply]
Thank you so much!
